inbm/dispatcher-agent/tests/unit/sota/test_sota.py

```diff
@@ -154,7 +155,8 @@ def test_run_pass(self, mock_run, mock_rollback_and_delete_snap, mock_print,
                 mock_rollback_and_delete_snap.assert_called_once()
             mock_run.assert_called_once()
         except SotaError as e:
-            self.assertEqual(str(e), "SOTA cache directory cannot be created")
+            assert (str(e) == "SOTA cache directory cannot be created" or
+                    str(e) == "Invalid OS or unable to detect OS for SOTA: Unsupported OS type or unable to detect OS")
```

Comment:
**Suggestion:** Replace the assertion with self.assertIn for consistency and better readability. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            except SotaError as e:
                self.assertIn(str(e), ["SOTA cache directory cannot be created",
                                       "Invalid OS or unable to detect OS for SOTA: Unsupported OS type or unable to detect OS"])
```